theseus/optimizer/nonlinear/tests/test_levenberg_marquardt.py

```diff
@@ -22,3 +32,34 @@ def test_levenberg_marquartd():
                 },
                 singular_check=damping < 0.001,
             )
+
+
+def test_ellipsoidal_damping_compatibility(mock_objective):
+    mock_objective.update({"v1": torch.ones(1, 1), "v2": torch.zeros(1, 1)})
+    for lsc in [th.LUDenseSolver, th.CholeskyDenseSolver]:
+        optimizer = th.LevenbergMarquardt(mock_objective, lsc)
+        optimizer.optimize(ellipsoidal_damping=True)
+        optimizer.optimize(damping_eps=0.1)
+
+    for lsc in [th.CholmodSparseSolver]:
+        optimizer = th.LevenbergMarquardt(mock_objective, lsc)
+        with pytest.raises(RuntimeError):
+            optimizer.optimize(ellipsoidal_damping=True)
+        with pytest.raises(RuntimeError):
+            optimizer.optimize(damping_eps=0.1)
+
+
+@pytest.mark.cuda
+def test_ellipsoidal_damping_compatibility_cuda(mock_objective):
+    mock_objective.to(device="cuda", dtype=torch.double)
+    mock_objective.update(
+        {
+            "v1": torch.ones(1, 1, device="cuda", dtype=torch.double),
+            "v2": torch.zeros(1, 1, device="cuda", dtype=torch.double),
+        }
+    )
+    for lsc in [th.LUCudaSparseSolver]:
+        optimizer = th.LevenbergMarquardt(mock_objective, lsc)
+        optimizer.linear_solver.reset(batch_size=1)
```

Comment:
Somehow I had missed that we need to call `solver.reset(batch_size=)` on `LUCudaSparseSolver` before running optimization. I think we want to hide this under the hood, but not sure what the best place to put this would be. @maurimo, Is this a function that can be called with different `batch_size` values every time? Would it make sense to have `NonlinearOptimizer.optimize()` always call this  function at the beginning (likewise for `LinearOptimizer.optimize()`.

Reply:
"reset is not necessary if the Solver is created with a valid linearization (if so reset is called automatically in the constructor, reset is available for tests and because of the limitations of Cuda api).
Keep in mind that every time reset is called the symbolic decomposition is recomputed. This is consequence of the limitations of the CuSolver-Refactor API, additionally to the issue of symbolic/numeric decompositions being the same the batch size is not flexible therefore it's not possible to create the sparse solver without knowing the size of the batch beforehand.
Therefore I would call "reset" only if it's necessary to change the batch size (or the number of contexts).

Reply:
"reset is not necessary if the Solver is created with a valid linearization (if so reset is called automatically in the constructor, reset is available for tests and because of the limitations of Cuda api).
Keep in mind that every time reset is called the symbolic decomposition is recomputed. This is consequence of the limitations of the CuSolver-Refactor API, additionally to the issue of symbolic/numeric decompositions being the same the batch size is not flexible therefore it's not possible to create the sparse solver without knowing the size of the batch beforehand.
Therefore I would call "reset" only if it's necessary to change the batch size (or the number of contexts).

Reply:
looks good to me!